src/utils/requestUtils.ts

```diff
@@ -48,4 +50,42 @@ export namespace requestUtils {
             stream.pipe(fse.createWriteStream(filePath).on('finish', resolve).on('error', reject));
         });
     }
+
+    export async function getFunctionAppMasterKey(azureSession: AzureSession, resourceId: string, accessToken: any): Promise<HttpOperationResponse> {
+        const subscriptionId: string = getSubscriptionFromId(resourceId);
+        const subscriptionContext: ISubscriptionContext = {
+            credentials: azureSession.credentials2,
+            subscriptionDisplayName: '',
+            subscriptionId: subscriptionId,
+            subscriptionPath: `/subscriptions/${subscriptionId}`,
+            tenantId: azureSession.tenantId,
+            userId: azureSession.userId,
+            environment: azureSession.environment,
+        };
+        const client: ServiceClient = await createGenericClient(subscriptionContext);
+        const request: WebResource = new WebResource();
+        request.prepare({
+            method: 'POST',
+            url: `https://management.azure.com/${resourceId}/host/default/listkeys?api-version=2018-11-01`,
```

Comment:
is it okay to hardcode the management.azure.com here?

Reply:
Yes, you are right. We don't need to hardcode. Just send `/${resourceId}/host/default/listkeys?api-version=2018-11-01` as the url and the clien t constructs it properly

Reply:
Will update in next iteration